fix(helm):  Display debug release info when upgrade failed because of timeout. Closes #3020

splisson opened the pull request:
When an upgrade fails because of a timeout waiting for it to be ready, in debug mode, output the release content.

Closes #3020

Commit:
build helm binary in tiller image

mparry committed:
(fix) Handle caFile alone being set for repos

Previously it was only respected if certFile and keyFile were also
specified. However, these are independent features.

splisson committed:
fix(helm):  When waiting for a release to be ready fails and in debug mode, output the release content Closes #3020

Commit:
Adding --set-string flag to force string values.

adshmh committed:
fix(helm): remove duplication in tests of repo-add command

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

adshmh committed:
fix(helm): refactor tests on helm dependency list command to remove duplication

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

robbert229 committed:
fixed an issue in versioning.mk (#3653)

The MUTABLE_IMAGE_RUDDER variables was using the DOCKER_VERSION instead of MUTABLE_VERSION
```
MUTABLE_IMAGE        := ${DOCKER_REGISTRY}/${IMAGE_PREFIX}/${SHORT_NAME}:${MUTABLE_VERSION}
MUTABLE_IMAGE_RUDDER := ${DOCKER_REGISTRY}/${IMAGE_PREFIX}/${SHORT_NAME_RUDDER}:${DOCKER_VERSION}
```

Commit:
Merge pull request #3695 from bacongobbler/windows-path-issues

fix windows path issues on `helm template -x`

adshmh committed:
fix output leak from tiller release install test

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

Commit:
Merge pull request #3699 from adshmh/fix-duplication-of-test-code-on-helm-repo-add

Refactor tests of repo-add command to remove duplication

Commit:
Merge pull request #3705 from adshmh/fix-duplication-of-test-code-on-helm-dependency

Refactor tests on helm dependency list command to remove duplication

Commit:
Merge pull request #3688 from adshmh/fix-duplication-of-test-code-on-helm-search

Fix helm search command error on index search failures

Commit:
ref(cmd/upgrade): update reuse-values flag descrip

helps clarify behavior. See #3655

Commit:
fix `helm get manifest` context deadline exceeded error

lestrrat committed:
Add actionable message in the error

See #3697

Commit:
Merge pull request #3652 from rajatjindal/history-format

add support for output format in json or yaml

adamreese committed:
fix(helm): fix helm history unit tests

fixes #3652

Commit:
Merge pull request #3677 from stuartleeks/patch-1

Fix link to github issues in CONTRIBUTING.md

andriisoldatenko committed:
Fixed typo

Commit:
Merge pull request #3719 from andriisoldatenko/main/typo-in-comments

Fixed typo

adamreese committed:
Merge pull request #3717 from adamreese/fix/history-output

fix(helm): fix helm history unit tests

eyalbe4 committed:
Authentication support for remote charts repositories (#3206)

Authentication support for remote charts repositories.

Commit:
Merge pull request #3599 from arturo-c/master

Adding --set-string flag to force string values.

alexey-igrychev committed:
Implement before-hook-creation delete policy

Existing helm.sh/hook-delete-policy annotation variables (hook-failed, hook-succeeded) do not allow to leave failed jobs for debugging without blocking the next job launching: every failed job must be deleted manually before the next related release is launching (installing, updating or rolling back).

New policy, before-hook-creation, removes the hook from previous release if there is one before the new hook is launched and can be used with another variable.

distorhead committed:
Add hook-delete-policy ReleaseServer unit tests

Cover hook-succeeded, hook-failed and before-hook-creation hook delete policies.

helgi committed:
Create index.yaml if missing when running repo index --merge

Fixes #3682

jstrachan committed:
add a link to Jenkins X and how it uses helm

for promoting applications through environments via GitOps

Ferenc- committed:
Fix HTML parsing when setting TAG with wget

Now the parsing part is the same as in the curl scenario

Commit:
Merge pull request #3737 from Ferenc-/master

Fix HTML parsing when setting TAG with wget

Commit:
Merge pull request #3716 from lestrrat/master

Add actionable message in the error

fqsghostcloud committed:
fix Syntax error

fix Syntax error

Commit:
Merge pull request #3738 from fqsghostcloud/fix-md-error

fix Syntax error

jascott1 committed:
Add issue_template.md which asks for version info.

jascott1 committed:
Update to specify version command output.

jascott1 committed:
Merge pull request #3743 from jascott1/issue_template

Add issue_template.md which asks for version info.

derekbassett committed:
Add App Version to the helm ls command.

jstrachan committed:
fixed typo + moved into Helm Included section

Commit:
Merge pull request #3733 from jstrachan/patch-2

add a link to Jenkins X and how it uses helm

Commit:
Merge pull request #3730 from helgi/feature/gh-3682

Create index.yaml if missing when running repo index --merge

Commit:
Merge pull request #3661 from bacongobbler/fix-default-connection-timeout

add back the default 5 second connection timeout

rmorrise committed:
Updated to mention the --purge flag on delete

See: https://github.com/kubernetes/helm/issues/2332

Commit:
Merge pull request #3750 from rmorrise/patch-1

Updated to mention the --purge flag on delete

cameronconradt committed:
Update files.go

Should suppress the warnings about the naming of these two functions. Added TODO

AdamDang committed:
Update tiller_ssl.md

"that" is duplicated.

Commit:
Merge pull request #3759 from AdamDang/patch-1

Update tiller_ssl.md

adamreese committed:
ref(*): kubernetes v1.10 support

edgarlanting committed:
Updated kubernetes_distros.md with DC/OS note.

Updated kubernetes_distros.md with a note that Helm is running fine on Kubernetes on DC/OS

flimzy committed:
Remove outdated note.

Commit:
Merge pull request #3771 from flimzy/doc

Remove outdated note from doc

adshmh committed:
Fix several golint warnings (#3756)

* fix(helm): fix golint warning due to ApiVersionV1 constant name

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

* fix(helm): fix golint warning due to ResolveChartVersionAndGetRepo comment

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

* fix(helm): fix golint warnings on HttpGetter type and SetCredentials method missing a comment

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

* fix(helm):fix golint warning due to comment on FindChartInAuthRepoURL function

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

* fix(helm): fix golint warning due to RepoFile type name

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

* fix(helm): fix golint warning due to ParseString comment

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

muhongwei committed:
fix 'mulitenant'->'multitenant' (#3755)

muhongwei committed:
fix 'eveything' -> 'everything' (#3754)

adamreese committed:
Merge pull request #3721 from adamreese/ref/kube-1.10

ref(*): kubernetes v1.10 support

edgarlanting committed:
Removed typos (additional plus signs)

Removed typos (additional plus signs)

Commit:
Merge pull request #3768 from edgarlanting/patch-1

Updated kubernetes_distros.md with DC/OS note.

Commit:
stream releases when listing

goshlanguage committed:
return a non 0 exit code when lint fails due to missing Chart.yaml

jkoleszar committed:
Fix tiller deployment on RBAC clusters

Adds automountServiceAccountToken when a serviceAccount is specified.

Prior to this, tiller falls back to contacting the KUBERNETES_SERVICE on
localhost:8080 rather than respecting the cluster IP in the
KUBERNETES_SERVICE_{HOST,PORT} environment variables.

Fixes #3460, fixes #3467.

adshmh committed:
fix(helm): fix output leak from unit tests of helm create command

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

Commit:
Merge pull request #3789 from adshmh/fix-output-leak-from-create-command-unit-tests

fix output leak from unit tests of helm create command

AdamDang committed:
Update rbac.md

In the main text and paragraph title, "Tiller" and "tiller" are both used, It's better use same format.

ReSearchITEng committed:
closes #3795

closes #3795

Commit:
release list paritioning

adshmh committed:
fix(helm) refactor release_testing unit tests to utilize runReleaseCases

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

fibonacci1729 committed:
Merge pull request #3782 from fibonacci1729/master

stream releases when listing

Commit:
Removes unnecessary if block

Commit:
import client auth plugins

Commit:
Merge pull request #3540 from flant/hook_recreate_delete_policy

Implement before-hook-creation delete policy

Commit:
Merge pull request #3817 from bacongobbler/fix-3806

import client auth plugins

Commit:
Fix #3822

thomastaylor312 committed:
Merge pull request #3799 from ReSearchITEng/master

closes #3795

thomastaylor312 committed:
Merge pull request #3813 from usabilla/merge-values

Removes unnecessary if block

thomastaylor312 committed:
Merge pull request #3784 from jkoleszar/tiller-rbac

Fix tiller deployment on RBAC clusters

AdamDang committed:
Typo fix helm->Helm

goshlanguage committed:
using existing mechanism to flag for failures

stuartleeks committed:
Switch flag to update existing record

Commit:
ref(cmd/helm): show grpc error msg from prettyError

Commit:
Merge pull request #3830 from michelleN/prettyError

ref(cmd/helm): show grpc error msg from prettyError

Commit:
fix(pkg/tiller): reuseValues combines all prev val

Resolves #3655
We were seeing that when running helm upgrade with the reuse-values
flag enabled that you could end up in the position where overrides
a.k.a computed values from previous revisions were not being saved on
the updated revision. This left us in a weird position where some
computed values would disappear mysteriously in the abyss. That
happened because computed values from previous revisions weren't merged
with the new computed values every time the reuse-values flag was used.
This PR merges computed values from the previous revisions so you don't
end up in that kind of conundrum.

Commit:
ref(pkg/tiller): clarify reuseValues comment

Commit:
Merge pull request #3713 from michelleN/reuse-values-docs

ref(cmd/upgrade): update reuse-values flag descrip

Commit:
Merge pull request #3824 from zdw/master

Fix #3822

Commit:
Merge pull request #3831 from michelleN/fix-3655

fix(pkg/tiller): saves all previous computed values on reuseValues

dwalleck committed:
Fixes typos introduced in #3540. Closes #3823

Commit:
Merge pull request #3832 from dwalleck/fix-typos-3823

Fixes typos introduced in #3540.

Commit:
rename TPR to CRD

thomastaylor312 committed:
Merge pull request #3794 from AdamDang/patch-2

Update rbac.md

stefanhenseler committed:
Updates readme with choco install command

thomastaylor312 committed:
Merge pull request #3783 from RyanHartje/fail_lint_when_chart_yaml_missing

return a non 0 exit code when lint fails due to missing Chart.yaml

Commit:
Merge pull request #3838 from synax/windows-choco-install

Updates readme with chocolatey install command for Windows

consideRatio committed:
toYaml - Fixes #3410 - trailing \n issue

`toYaml` utilized by `.Files` was introducing a new line. It is an issue since the new line is part of a functions output, it can't be whitespace chomped away so it would require a `trimSuffix "\n"` pipe. This commit trims one trailing `\n` from the toYaml output.

consideRatio committed:
toYaml - Fixes #3470 - trailing \n issue

`toYaml` was introducing a new line. It is an issue since the new line is part of a functions output, it can't be whitespace chomped away so it would require a `trimSuffix "\n"` pipe. This commit trims one trailing `\n` from the toYaml output.

consideRatio committed:
Updated tests for PR #3837

AdamDang committed:
Fix some typos

outout->output
Get's->get
the the->the

wind0r committed:
remove unused servicePort from default ingress template

Commit:
Merge pull request #3471 from adshmh/add-set-and-values-options-to-helm-package

feat: add --set and --values options to 'helm package'

thomastaylor312 committed:
Merge pull request #3843 from wind0r/remove_unuused_servicePort

remove unused servicePort from default ingress template

Commit:
Merge pull request #3836 from bacongobbler/rename-to-crd

rename TPR to CRD

thomastaylor312 committed:
Merge pull request #3829 from stuartleeks/set-rollback-failure

Fix: Failed rollback left in PENDING_ROLLBACK state

SlickNik committed:
Merge pull request #3709 from adshmh/fix-output-leak-of-tiller-release-install-test

fix output leak from tiller release install test

thomastaylor312 committed:
fix(package): Adds missing `set-string` flag and parameter

A recent PR (#3471) wasn't up to date with the latest master changes.
This fixes the issue and adds the right number of parameters to the
`vals` function call.

thomastaylor312 committed:
Merge pull request #3848 from thomastaylor312/fix/missing-package-flag

fix(package): Adds missing `set-string` flag and parameter

thomastaylor312 committed:
fix(docs): Add the missing docs

Because I was too stupid to regen the docs and flake made it so we
didn't get a reminder

Commit:
Merge pull request #3849 from thomastaylor312/fix/docs

fix(docs): Add the missing docs

AdamDang committed:
typo fix get->Get's

typo fix get->Get's

Commit:
Merge pull request #3842 from AdamDang/patch-1

Fix some typos

adshmh committed:
fix(helm) refactor reset command unit tests to remove duplication in test code

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

consideRatio committed:
Typo fix

`*/ }}` would cause an error actually, one must use `*/}}` or `*/ -}}`. Not obvious at all.

Commit:
Merge pull request #3856 from consideRatio/patch-1

Template comment in docs - syntax error fix

adshmh committed:
fix(helm): refactor tiller release install unit tests using chart and install request stubs

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

AdamDang committed:
Proper none not in capitals

Helm is a proper none, should be written in capitals.

Commit:
Merge pull request #3867 from AdamDang/patch-3

Proper none not in capitals

Commit:
Merge pull request #3715 from bacongobbler/fix-get-manifest

fix `helm get manifest` context deadline exceeded error

xianlubird committed:
Fix some typo

Signed-off-by: xianlubird <xianlubird@gmail.com>

Commit:
Merge pull request #3869 from xianlubird/features/typo

Fix some typo

eyalbe4 committed:
Basic auth credentials from repo not used in install/upgrade/fetch commands #3858

Commit:
Merge pull request #3871 from eyalbe4/master

Basic auth credentials from repo not used in install/upgrade/fetch co…

mrueg committed:
Remove Mercurial build-time dependency

Since the only dep (bitbucket.org/ww/goautoneg) that required it is gone in
https://github.com/kubernetes/helm/commit/845e99014362983d01179a411ae664b104e6c1f7#diff-f16a80eae23d5b298c2652448ec420cfL4

Commit:
Merge pull request #3872 from mrueg/remove-hg

Remove Mercurial build-time dependency

Commit:
Merge pull request #3837 from consideRatio/toYaml-pr

toYaml - Fix #3470 and #3410's trailing \n issues

Commit:
Merge pull request #3808 from adshmh/refactor-release-testing-unit-tests-to-utilize-runReleaseCases

refactor release_testing unit tests to utilize runReleaseCases

Commit:
remove optional field from charts and templates

Commit:
Update helm template -x to support children chart manifests in more cases

The previous code validated the manifests exist within the file system,
which only works when a chart is unpacked, meaning children charts
stored as tgz's in the parent chart's `charts/` directory don't exist as
individual files in the filesystem, causing validation to file.

Instead, we validate against the paths within the chart itself rather
than the filesystem, handling both chart.tgz's and charts that are
unpacked as directories in the chart/ directory.

thomastaylor312 committed:
Merge pull request #3853 from adshmh/refactor-reset-unit-tests-to-remove-duplication

refactor reset command unit tests to remove duplication

Commit:
Merge pull request #3876 from bacongobbler/fix-3780

remove optional field from charts and templates

Commit:
Merge pull request #3860 from adshmh/refactor-tiller-release-install-unit-tests-using-stubs

refactor tiller release install unit tests using chart and install request stubs

Commit:
Add working template to mariner chart for use in testing subcharts

Commit:
test helm template -x against subcharts stored as tgz's

Commit:
Disambigutate chartPaths variables in helm template tests

Commit:
Correctly use subtests in helm template tests

The testing.T in the subtest wasn't being used, so subtests were
effectively not being used. Additionally, in case subtests run in
parallel, we assign tt := tt to fix closure captures on range variables,
since the variables produced by a range are re-used between iterations.

Commit:
test helm template -x with non-existent manifest

SlickNik committed:
Merge pull request #3877 from chancez/enhance_helm_template_handling

Update helm template -x to support children chart manifests in more cases

Commit:
Merge pull request #3258 from mparry/fix-ca-file-option

(fix) Handle caFile alone being set for repos

jascott1 committed:
Merge pull request #3323 from jascott1/2620_importvals_warning_fix

fix(helm): fix importValues warnings from disabled charts

Jesse-Weinstein-Zonar committed:
Document apiVersion field

Added in https://github.com/kubernetes/helm/issues/1264

Commit:
Merge pull request #3892 from Jesse-Weinstein-Zonar/document_chart_apiVersion

Document apiVersion field

adshmh committed:
fix(helm): resolve linter's warning on template command unit tests

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

Commit:
Merge pull request #3897 from adshmh/fix-linter-warning-on-template-unit-test

Resolve warning from golint on template command unit test

Serg1i committed:
Create PodSecurityPolicy before Pods and ServiceAccounts

Commit:
ref(pkg/plugin): create clean path for extracting plugins

Commit:
Merge pull request #3899 from Serg1i/feat/kindSorter-add-PodSecurityPolicy

Create PodSecurityPolicy before Pods and ServiceAccounts

adshmh committed:
feat(helm): support removing multiple repositories with repo remove command

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

AdamDang committed:
Typo fix in functions_and_pipelines.md

useable->usable

Commit:
Revert "feat: add --set and --values options to 'helm package'"

This reverts commit a930eb7ff4dcba13831174496dda2445e81ebeaa.

Commit:
Merge pull request #3904 from bacongobbler/revert-3471

Revert "feat: add --set and --values options to 'helm package'"

AdamDang committed:
Correct the returned message

Correct the returned message

AdamDang committed:
Correct the returned message in reset_test.go

Correct the returned message

Commit:
Merge pull request #3905 from AdamDang/patch-5

Correct the returned message

Commit:
Merge pull request #3906 from AdamDang/patch-7

Correct the returned message in reset_test.go

Commit:
fix(pkg/strvals): evaluate "null" values

resolves #3891
Co-authored-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
Merge pull request #3912 from michelleN/process-null

 fix(pkg/strvals): evaluate "null" values

adamreese committed:
fix(kube): use correct object type in watch

fixes #3902

adamreese committed:
Merge pull request #3913 from adamreese/fix/3902-helm-test

fix(kube): use correct object type in watch

Commit:
swallow the error when returning the default HTTP client

Commit:
Merge pull request #3898 from michelleN/filepath

ref(pkg/plugin): create clean path for extracting files

Commit:
Merge pull request #3901 from AdamDang/patch-4

Typo fix in functions_and_pipelines.md

Commit:
Merge pull request #3903 from derekbassett/helm-ls-modification-rebase

Add App Version to the helm ls command.

Commit:
Merge pull request #3900 from adshmh/3886-support-removal-of-multiple-repos-from-cli

support removing multiple repositories using repo remove command

adamreese committed:
fix(kube): get correct versioned object from info helper

fixes: #3826

adamreese committed:
Merge pull request #3921 from adamreese/fix/3826-kube-version-objs

fix(kube): get correct versioned object from info helper

Commit:
Merge pull request #3915 from bacongobbler/fix-3881

swallow the error when returning the default HTTP client

colindickson committed:
fixed flag for tls ca cert option in the documentation

Commit:
Merge pull request #3929 from colindickson/master

fixed flag for tls ca cert option in the documentation

Commit:
Merge pull request #3692 from aleerizw/fix-sigsegv-create

Fixed SIGSEGV when running helm create with -p and no values.yaml file

AdamDang committed:
Typo fix in plugins.md "that that"->"that"

"that that"->"that"

AdamDang committed:
Update install.go

Commit:
Merge pull request #3941 from AdamDang/patch-8

Some typo fixes

jkammerl committed:
Avoid to call 'go' with empty -tags argument

go1.10.1 linux/amd64 complains: flag provided but not defined: -tags
when compiling with: "make build-cross dist". This seems to fix it.

databus23 committed:
Fix --tiller-namespace flag for plugins

This fixes using `--tiller-namespace $namespace` flag (without the equal sign) for helm plugins.

Commit:
Merge pull request #3944 from databus23/patch-4

Fix --tiller-namespace flag for plugins

adamreese committed:
fix(kube): output internal object table

fixes #3937

adamreese committed:
Merge pull request #3946 from adamreese/fix/3937-status

fix(kube): output internal object table

Commit:
Merge pull request #3942 from jkammerl/jkammerl-patch-2

Avoid to call 'go' with empty -tags argument

maratoid committed:
add --col-width to `helm search` (#3949)

* add customizeable --col-width

* make docs. whitespacing

BarryWilliams committed:
Changed whitespacing in comments

Commit:
Merge pull request #3950 from BarryWilliams/issue3747

Add comments to fix linter issues

jstoja committed:
Add quoting support in ingress to allow wildcard domain

eyalbe4 committed:
Fix for - Downloader plugins not used when downloading new repo's index.yaml #3938

jstoja committed:
Change tiller's Dockerfile to use USER nobody + upgrades to alpine:3.7

Commit:
remove need for type reflection

Commit:
Merge pull request #3968 from bacongobbler/fix-3962

fix(downloader): use plugin getters instead of a hardcoded HttpGetter

Commit:
Merge pull request #3965 from jstoja/dockerfile_root_user

Change tiller's Dockerfile to use USER nobody + upgrades to alpine:3.7

Commit:
Merge pull request #3956 from jstoja/support_ingress_wildcard

Add quoting support in ingress to allow wildcard domain

jstoja committed:
docs(helm): update Globs examples to work correctly

Commit:
Merge pull request #3970 from jstoja/docs_globs_examples

docs(helm): update Globs examples to work correctly

Commit:
#3763 replace backslash with forward slash

Commit:
Revert "toYaml - Fix #3470 and #3410's trailing \n issues"

Commit:
Revert "Fix tiller deployment on RBAC clusters"

Commit:
Refactor to be more precise

AdamDang committed:
Typo fix: evalutes->evaluates

Commit:
Merge pull request #3994 from AdamDang/patch-9

Typo fix: evalutes->evaluates

Commit:
Merge pull request #3988 from kubernetes/revert-3837-toYaml-pr

Revert "toYaml - Fix #3470 and #3410's trailing \n issues"

Commit:
Merge pull request #3990 from kubernetes/revert-3784-tiller-rbac

Revert "Fix tiller deployment on RBAC clusters"

AdamDang committed:
Typo fix: usa helm->use helm

usa helm->use helm

Commit:
Merge pull request #4018 from AdamDang/patch-1

Typo fix: usa Helm->use Helm

thomastaylor312 reviewed (approved):
This looks good to me. Sorry for the really long delay, this somehow slipped through the cracks in my notifications.

thomastaylor312 commented:
@bacongobbler Before I merge, do you think this is something we could bring into the next bug fix release?

splisson commented:
No problem. Thanks for your work maintaining helm

Sébastien

> On May 10, 2018, at 14:40, Taylor Thomas <notifications@github.com> wrote:
> 
> @thomastaylor312 approved this pull request.
> 
> This looks good to me. Sorry for the really long delay, this somehow slipped through the cracks in my notifications.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

thomastaylor312 reviewed (changes requested):
@splisson Hey, sorry to reset my review on this, but can you add this same logic to the `install` and `rollback` commands? It works great in `upgrade`, but it should have parity across the other commands as well

splisson committed:
Show release content when timeout occur for install and rollback. For rollback show it only when debug flag is set.

splisson committed:
Show release content when timeout occur for install and rollback. For rollback show it only when debug flag is set.

splisson committed:
Show release content when timeout occur for install and rollback. For rollback show it only when debug flag is set.

thomastaylor312 commented on cmd/helm/install.go:
```diff
@@ -264,23 +264,33 @@ func (i *installCmd) run() error {
-		return prettyError(err)
+
+	// If there is an error while waiting, make a call without waiting to get the release content
+	if (resp == nil || resp.Release == nil) && i.wait {
```
Should this be wrapped with `settings.Debug`?

thomastaylor312 replied:
Actually, I looked some more into this. I think it should be `i.printRelease` which is where it prints debug

thomastaylor312 commented:
Ok @splisson, now that I have my head on straight and am not trying to do multiple things at once, it looks like `rollback` never printed anything before and I thought it did. Also, the install needs to use `i.printRelease`, which isn't intuitive I know. So I would remove the change to rollback and make a slight adjustment to the `install` logic and then this is good to go. 

I apologize for all of the confusion, 100% my fault.

splisson commented:
The existing trace for sure install did not have it before 

Sébastien

> On May 10, 2018, at 16:48, Taylor Thomas <notifications@github.com> wrote:
> 
> @thomastaylor312 commented on this pull request.
> 
> In cmd/helm/install.go:
> 
> > @@ -264,23 +264,33 @@ func (i *installCmd) run() error {
>  		helm.InstallDisableHooks(i.disableHooks),
>  		helm.InstallTimeout(i.timeout),
>  		helm.InstallWait(i.wait))
> -	if err != nil {
> -		return prettyError(err)
> +
> +	// If there is an error while waiting, make a call without waiting to get the release content
> +	if (resp == nil || resp.Release == nil) && i.wait {
> Should this be wrapped with settings.Debug?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

thomastaylor312 commented:
Yeah, in `install` you shouldn't need the `if settings.Debug` block, just the call to `printRelease` should be switched to `i.printRelease`

splisson committed:
use i.printRelease

splisson committed:
use i.printRelease

technosophos committed:
feat(tiller): support CRD installation (#3982)

This adds support for installing CRDs well before any other resource
kinds are installed.

This PR introduces a new hook, `crd-install`, that fires before
manifests are even validated. It is used to install a CRD before any
other part of a chart is installed.

Currently, this hook is _only implemented for install_. That means we
currently cannot add new CRDs during `helm upgrade`, nor can they
be rolled back. This is the safest configuration, as the update/rollback
cycle gets very challenging when CRDs are added and removed.

gabrielsvinha committed:
Upgrade moniker version (#4034)

adamreese committed:
fix(kube): run schema validation on BuildUnstructured

winjer committed:
Support values files on remote servers with self-signed certs / client cert authentication (#4003)

* support tls for values file access

- uses --ca-file / --cert-file / --key-file

* also for upgrade command

adamreese committed:
Merge pull request #4042 from adamreese/fix/kube-validate

fix(kube): run schema validation on BuildUnstructured

rajatjindal committed:
fix lint warning

thomastaylor312 commented:
@bacongobbler Yeah, this has evolved since my original comment when I mentioned you 🙂

Commit:
replace with a link to the latest releases page

This removes a step required every time we release Helm, making it
simpler to cut a new release.

AdamDang committed:
Typo fix: mach->match

mach->match

Commit:
Merge pull request #4061 from AdamDang/patch-3

Typo fix: mach->match

jmeickle committed:
Don't request Tiller version in installer

This makes the script hang for quite a while if it can't connect to Tiller to retrieve this information that isn't being used anyways.

Commit:
Merge pull request #4062 from Eronarn/patch-1

Don't request Tiller version in installer

Commit:
Merge pull request #4052 from bacongobbler/latest-releases-page

docs(README): replace with a link to the latest releases page

Commit:
Merge pull request #4043 from rajatjindal/fix-lint-warn

fix lint warning

AdamDang committed:
Update capabilities.go

Commit:
Merge pull request #4069 from AdamDang/patch-4

Typo fix: KuberVerison->KuberVersion

yenicapotediaz committed:
removed --install option from --namespace.

balboah committed:
Add helm template --is-upgrade

Useful for toggling .Release.IsInstall / .Release.IsUpgrade which
in turn may cause templates to render differently.

balboah committed:
make docs

recipe for target 'verify-docs' failed

goshlanguage committed:
adding docs for GKE for helm 2

Commit:
Merge pull request #4116 from RyanHartje/4013

[4013] adding docs for GKE for helm 2

slayerjain committed:
docs(helm): Added double quotes to example

In the current version of helm (2.8.2), there needs to be double quotes around the value to support escaping special characters.

mf-lit committed:
Update tiller_ssl.md

Add -days parameter to openssl signing commands for clarity. Also add some related troubleshooting steps.

Commit:
Merge pull request #4121 from mf-lit/patch-1

Update tiller_ssl.md

Commit:
Merge pull request #4119 from slayerjain/patch-1

docs(helm): Added double quotes to example

Commit:
Merge pull request #4105 from balboah/feature/template_release_isinstall

Add helm template --is-upgrade

Commit:
Merge pull request #4104 from yenicapotediaz/master

removed --install option from --namespace to solve inconsistency with the `helm lint` documentation

Commit:
update security vulnerability email

Commit:
Merge pull request #4124 from michelleN/security-email

update security vulnerability email

Commit:
Merge pull request #3984 from muehlburger/dev

Fix filenames in tarball created on windows

radbaron committed:
Documentation: add syntax highlighting to code examples & fix spelling of kube primitives

Commit:
Merge pull request #1929 from bacongobbler/helm-image

build helm binary in tiller image

rsalmond committed:
small typo fix

Commit:
Merge pull request #4151 from rsalmond/typofix

small typo fix

mattjmcnaughton committed:
Fix test failure message

The error message we reported if the test failed was incorrect.

Commit:
Merge pull request #4153 from mattjmcnaughton/mattjmcnaughton/fix-test-failure-message

Fix test failure message

AdamDang committed:
Typo fix: retruns->returns

retruns->returns

eicnix committed:
Added documentation for the tpl function

Commit:
Merge pull request #4159 from AdamDang/patch-5

Typo fix: retruns->returns

Commit:
Merge pull request #4160 from eicnix/master

Added documentation for the tpl function

andrzejwp committed:
Add link to cdwv/awesome-helm

Commit:
Merge pull request #4175 from andrzejwp/patch-1

Add link to cdwv/awesome-helm

mirake committed:
Typo fix: indentifies -> identifies

Commit:
Merge pull request #4179 from mirake/fix-typos

Typo fix: indentifies -> identifies

Commit:
Merge pull request #4139 from radbaron/master

Documentation formatting [Tiny]

liyongxin committed:
typo fix for lengh to length

Commit:
Merge pull request #4196 from liyongxin/typo-fix-lengh-to-length

typo fix for lengh to length

bacongobbler commented:
ping

splisson committed:
Revert "Rebased"

This reverts commit 1737b711ca07cd1fa587ae70f1a0c9ff89dead6e.

splisson committed:
Resolved conflict

splisson committed:
Fix res => resp

splisson commented:
I added the change in install.go , Rebased and resolved conflicts. The test that is failing now is looking for a file I don't see in master on main repository neither

splisson committed:
Fix test to check if an error was returned (anything not nil)

splisson commented:
Looks ok now

bacongobbler commented:
hmm, looks like the rebase went bad actually. Could you please fix this PR? Thanks so much!

bacongobbler reviewed (changes requested):
needs a clean rebase

splisson commented:
the rebase included 200+ previous commits that happened between the time I did my last commit and now. Isn't that expected ?

splisson commented:
I'll prepare a fresh PR from a new branch

splisson commented:
Closing. Replaced by PR https://github.com/kubernetes/helm/pull/4207

splisson commented:
@bacongobbler I submitted a new PR #4207
